client/src/views/client-area/dashboard/CustomOrderProcessingPage.vue

```diff
@@ -2,14 +2,143 @@
 
 <template>
   <div class="client-page-processing">
-    <ms-report-text :theme="MsReportTheme.Info">
-      {{ $msTranslate('clientArea.dashboard.processing') }}
-    </ms-report-text>
+    <template v-if="!querying && !error && customOrderStatus && currentStatusStep !== CustomOrderStatusStep.Unknown">
+      <div class="process-container">
+        <div
+          class="process-step"
+          v-for="(step, index) in steps"
+          :key="index"
+          :class="{
+            'process-step-todo': currentStatusStep < step.status,
+            'process-step-active': currentStatusStep === step.status,
+            'process-step-done': currentStatusStep > step.status,
+          }"
+        >
+          <div class="process-step-icon">
+            <div
+              class="dot-todo"
+              v-if="currentStatusStep < step.status"
+            />
+            <div
+              class="dot-active"
+              v-if="currentStatusStep === step.status"
+            />
+            <ion-icon
+              class="process-step-icon__item"
+              :icon="checkmarkCircle"
+              v-if="currentStatusStep > step.status"
+            />
+          </div>
+          <div class="process-step-text">
+            <ion-text class="process-step-text__title title-h3">
+              {{ $msTranslate(step.title) }}
+            </ion-text>
+            <ion-text
+              class="process-step-text__info body"
+              v-if="currentStatusStep === step.status"
+            >
+              {{ $msTranslate(step.description) }}
+            </ion-text>
+          </div>
+        </div>
+      </div>
+    </template>
+    <template v-else-if="querying">
+      <ms-spinner />
+    </template>
+    <template v-else-if="error || currentStatusStep === CustomOrderStatusStep.Unknown">
+      <ms-report-text :theme="MsReportTheme.Error">
+        {{ $msTranslate(error) }}
+      </ms-report-text>
+    </template>
   </div>
 </template>
 
 <script setup lang="ts">
-import { MsReportText, MsReportTheme } from 'megashark-lib';
+import { IonIcon, IonText } from '@ionic/vue';
+import { checkmarkCircle } from 'ionicons/icons';
+import { BmsAccessInstance, CustomOrderStatusResultData, DataType, BmsOrganization, CustomOrderStatus } from '@/services/bms';
+import { isDefaultOrganization } from '@/views/client-area/types';
+import { ref, onMounted } from 'vue';
+import { MsSpinner, MsReportText, MsReportTheme } from 'megashark-lib';
+import { getCustomOrderStatusTranslationKey } from '@/services/translation';
+
+const customOrderStatus = ref<CustomOrderStatusResultData>();
+const querying = ref(true);
+const error = ref('');
+
+const props = defineProps<{
+  organization: BmsOrganization;
+}>();
+
+enum CustomOrderStatusStep {
```

Comment:
We already have an enum `CustomOrderStatus`, it's redundant to create another one and forces you to match them together. Instead if you just need to add some index for comparaison, you can do

```typescript
const CustomOrderStatusSteps = {
  [CustomOrderStatus.Unknown]: 0,
  [CustomOrderStatus.NothingLinked]: 1,
  ...
}
// and then we you want to get the value
CustomOrderStatusSteps[CustomOrderStatus.Unknown];
CustomOrderStatusSteps[step.status];
```